extensions/observability-devservices/deployment/src/main/java/io/quarkus/observability/deployment/ObservabilityDevServiceProcessor.java

```diff
@@ -250,14 +257,24 @@ public DevServicesResultBuildItem.RunningDevService get() {
                 .locateContainer(
                         capturedDevServicesConfiguration.serviceName(), capturedDevServicesConfiguration.shared(),
                         LaunchMode.current(), (p, ca) -> config.putAll(dev.config(p, ca.getHost(), ca.getPort())))
-                .map(new Function<String, DevServicesResultBuildItem.RunningDevService>() {
-                    @Override
-                    public DevServicesResultBuildItem.RunningDevService apply(String cid) {
-                        log.infof("Dev Service %s re-used, config: %s", devId, config);
-                        return new DevServicesResultBuildItem.RunningDevService(Feature.OBSERVABILITY.getName(), cid,
-                                null, config);
-                    }
+                .map(cid -> {
+                    log.infof("Dev Service %s re-used, config: %s", devId, config);
+                    return new DevServicesResultBuildItem.RunningDevService(Feature.OBSERVABILITY.getName(), cid,
+                            null, config);
                 })
+                .or(() -> ComposeLocator.locateContainer(composeProjectBuildItem,
+                        List.of(capturedDevServicesConfiguration.imageName()), LaunchMode.current())
+                        .stream().findFirst()
+                        .map(r -> {
+                            Map<String, String> cfg = new LinkedHashMap<>();
+                            for (ContainerInfo.ContainerPort port : r.containerInfo().exposedPorts()) {
+                                cfg.putAll(dev.config(port.privatePort(),
```

Comment:
@alesj could we use these properties to setup the extensions instead of relying on the internal system properties?

Reply:
@brunobat I think we can make adjustments in later iterations..

Reply:
@brunobat wdym? On which `internal system properties` do we rely on?

Reply:
@alesj @brunobat  we could use the `service-name` attribute. if it would've been `lgtm` instead of `quarkus`.
Only keycloak and lgtm have `quarkus` as service name. I'd like to change that, if you're ok.

Reply:
It makes sense

Reply:
@brunobat I don't see (yet) how the properties you mention can replace the current "internal" ones ...